src/relax/transform/infer_layout_utils.h

```diff
@@ -0,0 +1,201 @@
+ */
+using FRelaxInferLayout = runtime::TypedPackedFunc<InferLayoutOutput(
+    Call call, Map<String, Array<String>> desired_layouts, VarLayoutMap var_layout_map)>;
+
+/*!
+ * \brief Initialize a layout given the number of dimensions.
+ * \param ndim The number of dimensions.
+ * \return The initialized layout.
+ */
+Layout InitialLayout(size_t ndim);
+
+/*!
+ * \brief Initialize a nested layout given the struct info.
+ * \param sinfo The sinfo.
+ * \return The initialized nested layout.
+ */
+NLayout InitialNLayout(const StructInfo& sinfo);
+
+/*!
+ * \brief Initialize a nested layout given expression
+ * \param sinfo The expr
+ * \return The initialized nested layout.
+ */
+NLayout InitialNLayout(const Expr& expr);
+
+/*!
+ * \brief Transpose the input layout like the src layout to the dst layout.
+ * \param input The input layout.
+ * \param src The source layout.
+ * \param dst The destination layout.
+ * \return The transposed input layout.
+ */
+Layout TransposeLike(const Layout& input, const Layout& src, const Layout& dst);
+
+/*!
+ * \brief Transpose the input string like the src layout to the dst layout.
+ * \param input The input str.
+ * \param src The source layout.
+ * \param dst The destination layout.
+ * \return The transposed input str.
+ */
+String TransposeStrLike(const String& input, const Layout& src, const Layout& dst);
+
+/*!
+ * \brief Find axis in the dst layout.
+ * \param dst The destination layout.
+ * \param axis The axis to be found
+ * \return The axis in the dst layout.
+ */
+int TransposeAxis(const Layout& dst, int axis);
+
+/*!
+ * \brief Get the layout of the expr. The expr must be a Tensor.
+ * \param var_layout_map The layout of the variables.
+ * \param arg The expr.
+ * \return The layout of the expr.
+ */
+Layout GetLayout(VarLayoutMap var_layout_map, const Expr& arg);
+
+/*!
+ * \brief Get the nested layout of the expr. The expr must be a nested Tensor.
+ * \param var_layout_map The layout of the variables.
+ * \param arg The expr.
+ * \return The nested layout of the expr.
+ */
+NLayout GetNLayout(VarLayoutMap var_layout_map, const Expr& arg);
+
+/*!
+ * \brief Check if the op is not in the desired layout
+ * \param call The call node contains the op
+ * \param desired_layouts The desired layouts of the operator.
+ * \return True if the op is not in the desired layout.
+ */
+bool NoDesiredLayout(Call call, Map<String, Array<String>> desired_layouts);
+
+/*!
+ * \brief Default layout conversion infer function, which simply requires original
+ * layout of the call.
+ * \param call The call node.
+ * \param desired_layouts The desired layouts of the operator.
+ * \param var_layout_map The layout of the variables.
+ * \return The layout conversion infer result.
+ */
+InferLayoutOutput InferLayoutDefault(const Call& call,
+                                     const Map<String, Array<String>>& desired_layouts,
+                                     VarLayoutMap var_layout_map);
```

Comment:
I suppose that the default one is no longer used since we use `Optional<InferLayoutOutput>` on the pass side.